packages/nimble-components/src/rich-text-mention/base/view/index.ts

```diff
@@ -0,0 +1,34 @@
+import { FoundationElement } from '@microsoft/fast-foundation';
+import { attr } from '@microsoft/fast-element';
+
+/**
+ * The base class for Mention View Node
+ */
+export class RichTextMentionView extends FoundationElement {
+    /**
+     * Stores the unique URL of the mentioned user matching the pattern
+     *
+     * @public
+     * HTML Attribute: mention-href
+     */
+    @attr({ attribute: 'mention-href' })
+    public mentionHref?: string;
```

Comment:
I saw in the AzDO HLD that the proposal is to render the user ID if the user name hasn't loaded yet. Is that accurate? If so, what layer will be responsible for that? I could imagine the view doing it, though it would need access to the ID string but right now it has the full href so it would need to extract the ID.

This should probably be added in a follow-on PR, but it makes me question what the purpose of the `mention-href` attribute is. I don't see it used for anything. If it doesn't have a clear planned purpose then I'd vote to remove it until we implement that feature.

Reply:
Yes, the user ID will be in place of the name if the mapping element is not given and also if the link in the markdown matches the pattern correctly. I also mentioned it in the Nimble HLD [here](https://github.com/ni/nimble/blob/main/packages/nimble-components/src/rich-text/specs/mention-hld.md#user-mention-view-visible-ui-element:~:text=the%20mentioned%20user-,mention%2Dlabel,-%3A%20string%20%2D%20the%20user) for your reference.

> what layer will be responsible for that?

This will be handled only in the markdown-parser if the given markdown string contains mention node (in autolink format) that matches the pattern. For example, if `<users:1>` markdown string is given as an input and the pattern is also the same as `users:.*` then the `@1` will be shown by updating the `mention-label` of this element in the parser.

Once the mapping element is updated and the `users:1` name is provided the ID will be replaced with the mapped name by getMarkdown + setMarkdown in the editor.

> what the purpose of the `mention-href` attribute is.

It is used only in the markdown serialization where we get the current state of the Tiptap, read the `mention-href` attribute value of this mention view node, and construct the markdown string.

Reply:
Yea this stuff is complex, it's used for the copy-paste workflow. When you copy and paste a chunk of view between rich text editors the rendered mention has the mention-href that goes into the system and is treated as a link node and then should go down the same code path as mentions for links.

Makes me think of a test case of copying between rich comments that have different mention features  / no mention features configured. The mentions copied should just become links?

And I'm only 80% sure that's accurate. I think it would be preferable on copy to turn the mention views to anchors so that stuff never is dealt with externally. But I don't think that was how the prototypes behaved. And I know keyboards can get multiple representations, not sure how tiptap is dealing with that.

Reply:
> The mentions copied should just become links?

Currently, we don't know how to copy just a link when mention node is copied. In the prototypes, we modified the content only when pasting the node as you said. However, when quickly surfing around the Prosemirror we could find some ways to transform the copied content like [transformCopied](https://prosemirror.net/docs/ref/#view.EditorProps.transformCopied). When we implement it, we will see how can we copy just a link at least in the editor. 

But for the viewer, since it is not using any external libraries like Prosemirror/Tiptap to render the content in view, we might have to look for some native options to restrict copying the node.

Reply:
Both of those sound like potentially useful future enhancements assuming they aren't considered requirements for the first release (which seems unlikely giving the time pressure)